tutorials/high_performance_networking/README.md

```diff
@@ -0,0 +1,2179 @@
+2. The `advanced_network` section is passed to the `AdvNetworkOpRx` and `AdvNetworkOpTx` operators which are responsible for setting up the NIC.
+3. `manager` is the backend networking library. default: `dpdk`. Other: `gpunetio` (DOCA GPUNet IO + DOCA Ethernet & Flow). Coming soon: `rivermax`, `rdma`.
+4. `master_core` is the ID of the CPU core used for setup. It does not need to be isolated, and is recommended to differ differ from the `cpu_core` fields below used for polling the NIC.
+5. The `memory_regions` section lists where the NIC will write/read data from/to when bypassing the OS kernel. Tip: when using GPU buffer regions, keeping the sum of their buffer sizes lower than 80% of your BAR1 size is generally a good rule of thumb 👍.
+6. A descriptive name for that memory region to refer to later in the `interfaces` section.
+7. The type of memory region. Best options are `device` (GPU), or `huge` (pages - CPU). Also supported but not recommended are `malloc` (CPU) and `pinned` (CPU).
+8. The GPU ID for `device` memory regions. The NUMA node ID for CPU memory regions.
+9. The number of buffers in the memory region. A higher value means more time to process the data, but it takes additional space on the GPU BAR1. Too low increases the risk of dropping packets from the NIC having nowhere to write (Rx) or the risk of higher latency from buffering (Tx). Need a rule of thumb 👍? 3x the `batch_size` below is a good starting point.
+10. The size of each buffer in the memory region. These should be equal to your maximum packet size, or less if breaking down packets (ex: header data split, see the `rx` queue below).
+11. The `interfaces` section lists the NIC interfaces that will be configured for the application.
+12. A descriptive name for that interface, currently only used for logging.
+13. The PCIe/bus address of that interface, as identified in previous sections.
+14. Each interface can have a `tx` (transmitting) or `rx` (receiving) section, or both if you'd like to configure both Tx and Rx on the same interface.
+15. The `queues` section lists the queues for that interface. Queues are a core concept of NICs: they handle the actual receiving or transmitting of network packets. Rx queues buffer incoming packets until they can be processed by the application, while Tx queues hold outgoing packets waiting to be sent on the network. The simplest setup uses only one receive and one transmit queue. Using more queues allows multiple streams of network traffic to be processed in parallel, as each queue can be assigned to a specific CPU core, and are assigned their own memory regions that are not shared.
+16. A descriptive name for that queue, currently only used for logging.
+17. The ID of that queue, which can be referred to later in the `flows` section.
+18. The number of packets per batch. The `advanced_network` Rx and Tx operators will "tick" for each batch.
```

Comment:
The TX technically doesn't care about this number when transmitting. The user just has to make sure they're not transmitting more than this

Reply:
I was wondering.

Why do we even have a limit for Tx? it can't just shoot to the NIC whatever our Tx operator sends it?

Reply:
Because the user still has to allocate a burst of data, and that burst is always bound by the maximum number of packets. We could extend the API such that instead they only allocate a smaller number, but I'm not sure if that would be useful.

RX burst are sent either when the number of packets matches the burst size, or when the timer expires.